front/components/agent_builder/capabilities/AddToolsDropdown.tsx

```diff
@@ -6,51 +6,117 @@ import {
   DropdownMenuContent,
   DropdownMenuItem,
   DropdownMenuLabel,
+  DropdownMenuSearchbar,
   DropdownMenuTrigger,
+  Spinner,
 } from "@dust-tt/sparkle";
-import type { FieldArray, FieldArrayWithId } from "react-hook-form";
+import assert from "assert";
+import { uniqueId } from "lodash";
+import { useState } from "react";
+import type { FieldArrayWithId, UseFieldArrayAppend } from "react-hook-form";
 
-import type { AgentBuilderFormData } from "@app/components/agent_builder/AgentBuilderFormContext";
-import type { AgentBuilderAction } from "@app/components/agent_builder/types";
-import { getDataVisualizationActionConfiguration } from "@app/components/assistant_builder/types";
+import type {
+  AgentBuilderFormData,
+  AgentBuilderMCPAction,
+} from "@app/components/agent_builder/AgentBuilderFormContext";
+import type { MCPServerViewTypeWithLabel } from "@app/components/agent_builder/MCPServerViewsContext";
+import type { ActionSpecification } from "@app/components/agent_builder/types";
+import { getDefaultMCPAction } from "@app/components/agent_builder/types";
+import {
+  DEFAULT_DATA_VISUALIZATION_DESCRIPTION,
+  DEFAULT_DATA_VISUALIZATION_NAME,
+} from "@app/lib/actions/constants";
+import { getMcpServerViewDisplayName } from "@app/lib/actions/mcp_helper";
+import { getAvatar } from "@app/lib/actions/mcp_icons";
 import { DATA_VISUALIZATION_SPECIFICATION } from "@app/lib/actions/utils";
+import type { MCPServerViewType } from "@app/lib/api/mcp";
 
 interface AddToolsDropdownProps {
-  tools: FieldArrayWithId<AgentBuilderFormData, "actions">[];
-  addTools: (
-    value:
-      | FieldArray<AgentBuilderFormData, "actions">
-      | FieldArray<AgentBuilderFormData, "actions">[]
-  ) => void;
+  tools: FieldArrayWithId<AgentBuilderFormData, "actions", "id">[];
+  addTools: UseFieldArrayAppend<AgentBuilderFormData, "actions">;
+  setSelectedAction: React.Dispatch<
+    React.SetStateAction<AgentBuilderMCPAction | null>
+  >;
+  defaultMCPServerViews: MCPServerViewTypeWithLabel[];
+  nonDefaultMCPServerViews: MCPServerViewTypeWithLabel[];
+  isMCPServerViewsLoading: boolean;
+  dataVisualization: ActionSpecification | null;
 }
 
-export function AddToolsDropdown({ tools, addTools }: AddToolsDropdownProps) {
-  function onClickDataVisualization() {
-    const dataVisualizationConfig = getDataVisualizationActionConfiguration();
-    if (!dataVisualizationConfig) {
+export function AddToolsDropdown({
+  addTools,
+  setSelectedAction,
+  defaultMCPServerViews,
+  nonDefaultMCPServerViews,
+  dataVisualization,
+  isMCPServerViewsLoading,
+}: AddToolsDropdownProps) {
+  const [searchText, setSearchText] = useState("");
+  const [filteredServerViews, setFilteredServerViews] = useState([
+    ...defaultMCPServerViews,
+    ...nonDefaultMCPServerViews,
+  ]);
+  const [filteredDataViz, setFilteredDataViz] = useState(dataVisualization);
```

Comment:
This feels very stupid but at least logic is more or less the same as filteredServerViews, but I might regret it after

Reply:
Hmmm 😅 let's add a todo to ditch later ?

Reply:
How do we handle it in the assistant builder?